### DISCUSSION TITLE
How to print string in kerenl 4.4 version

### DISCUSSION THREAD

**Discussion by antiquerefer** [bpftrace/bpftrace#1984] [Q&A]
bpftrace -e 'uprobe:/lib/libsqlite3.so:sqlite3_exec { printf("%s\n", str(arg1)) }'

I want to use str() to print the string parameter.
However it said "invalid func unknown#45".

And I look it up a little bit.
Found out that str() is using BPF_FUNC_probe_read_str helper function.
And this helper function add in kernel version 4.11.

Is there convenient ways to print the string parameter in kernel version 4.4?

**Comment by fbs** [comment]
Sadly not really. What you can try is buf() but that doesn't check for Null so you'd have to filter the data later
